master branch build failure  with expanding macro `opts`

Please answer these questions before submitting your issue. Thanks!

1. What version of Rust are you using (`rustc --version`)?

rustc 1.25.0-nightly (73ac5d6a8 2018-01-11)

2. What operating system and processor architecture are you using?

Darwin lambda-cat.local 17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64

3. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.

make

4. What did you expect to see?

build success

5. What did you see instead?

```sh
lambda-cat.local ➜  tikv git:(master) ✗ make
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#protobuf:1.4.1
   Compiling tikv v2.0.0-rc.4.1 (file:///Users/LambdaCat/code/go/tikv)
error: recursion limit reached while expanding the macro `opts`
    --> src/util/rocksdb/engine_metrics.rs:1308:9
     |
1308 | /         register_int_gauge_vec!(
1309 | |             "tikv_engine_oldest_snapshot_duration",
1310 | |             "Oldest unreleased snapshot duration in seconds",
1311 | |             &["db"]
1312 | |         ).unwrap();
     | |_________^
     |
     = help: consider adding a `#![recursion_limit="200"]` attribute to your crate
     = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: Could not compile `tikv`.

To learn more, run the command again with --verbose.
make: *** [release] Error 101
lambda-cat.local ➜  tikv git:(master) ✗ git pull
warning: 重定向到 https://github.com/pingcap/tikv.git/
已经是最新的。
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -23,7 +23,7 @@
 #![cfg_attr(feature = "dev", feature(plugin))]
 #![cfg_attr(feature = "dev", plugin(clippy))]
 #![cfg_attr(not(feature = "dev"), allow(unknown_lints))]
-#![recursion_limit = "100"]
+#![recursion_limit = "200"]
 #![feature(ascii_ctype)]
 #![allow(module_inception)]
 #![allow(should_implement_trait)]

